<?xml version="1.0"?>
<launch>

  <group ns="PF">
    <rosparam file="$(find gaden_sims)/config/pfParams10x10.yaml" />
  </group>

  <arg name="RobotID"      default="1"/>
  <arg name="xminRaster" default="0"/>
  <arg name="xmaxRaster" default="10"/>
  <arg name="yminRaster" default="0"/>
  <arg name="ymaxRaster" default="10"/>
  <!-- <arg name="stayTime" default="0.25"/> -->

  <!-- Fake position publisher -->
  <node name="PF" pkg="gaden_sims" type="infoPFGaden.py" output="screen">
    <param name="xmin"     type="double"  value="$(arg xminRaster)"    /> # m
    <param name="xmax"     type="double"  value="$(arg xmaxRaster)"    /> # m
    <param name="ymin"     type="double"  value="$(arg yminRaster)"    /> # m
    <param name="ymax"     type="double"  value="$(arg ymaxRaster)"    /> # m
    <!-- <param name="stayTime" type="double"  value="$(arg stayTime)"/> -->
    <param name="RobotID"  type="double"  value="$(arg RobotID)" />
    <param name="printTimeSteps"          value="True"  />
  </node>

  <node name="copter_visualization" pkg="mavros_extras" type="visualizationCustom">
      <param name="num_rotors" value="4"/>
      <param name="arm_len" value="0.22"/>
      <!-- <param name="UAV_Name" value="$(arg UAV)"/> -->
      <param name="fixed_frame_id" value="map_gaden"/>
      <param name="child_frame_id" value="pid0_frame"/>
  </node>

</launch>
